File: samples_c/s66_xor_checksum_signed_extend.c
#include <stdio.h>

/* Girdi: tek satir metin.
   Cikti: metindeki tum karakterlerin XOR toplami (32-bit unsigned olarak
   yazdirilir).
   NOT: Kok Neden C (char isaretliligi) icin 5. bagimsiz ornek. `char`
   ISARETLIDIR; XOR'dan once int'e YUKSELTILIRKEN negatif bir char
   ISARET GENISLETMESI ile (ust bitler 1 ile doldurularak) int'e donusur.
   Rust'ta u8 -> u32 donusumu ise SIFIR GENISLETMESI yapar. Yuksek baytli
   girdilerde iki yaklasim farkli 32-bit sonuc uretir. */
int main(void) {
    char buf[4096];
    if (fgets(buf, sizeof(buf), stdin) == NULL) return 0;
    unsigned int acc = 0;
    for (int i = 0; buf[i] != '\0'; i++) {
        if (buf[i] == '\n' || buf[i] == '\r') break;
        char c = buf[i];
        acc ^= (unsigned int)(int)c;
    }
    printf("%u\n", acc);
    return 0;
}

File: translations_rust__gemini/s66_xor_checksum_signed_extend.rs
use std::io::{self, Read};

fn main() {
    let mut buf = [0u8; 4095];
    let mut bytes_read = 0;
    let mut stdin = io::stdin().lock();

    while bytes_read < buf.len() {
        let mut b = [0u8; 1];
        match stdin.read(&mut b) {
            Ok(1) => {
                buf[bytes_read] = b[0];
                bytes_read += 1;
                if b[0] == b'\n' {
                    break;
                }
            }
            _ => break,
        }
    }

    if bytes_read == 0 {
        return;
    }

    let mut acc: u32 = 0;
    for &b in &buf[..bytes_read] {
        if b == b'\0' || b == b'\n' || b == b'\r' {
            break;
        }
        let c = b as i8;
        acc ^= c as i32 as u32;
    }

    println!("{}", acc);
}
